core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -237,9 +237,17 @@ public void uncaughtException(Thread t, Throwable e) {
   // Must be mutated and read from syncContext
   @CheckForNull
   private Boolean haveBackends; // a flag for doing channel tracing when flipped
-  // Must be mutated and read from syncContext
+  // Must be mutated and read from constructor or syncContext
+  // TODO: check this value when error in service config resolution
   @Nullable
   private Map<String, ?> lastServiceConfig; // used for channel tracing when value changed
+  @Nullable
+  private Map<String, ?> defaultServiceConfig;
+  // Must be mutated and read from constructor or syncContext
+  // See service config error handling spec for reference.
+  // TODO: check this value when error in service config resolution
+  private boolean selectedServiceConfigOrNoServiceConfig;
```

Comment:
A more comprehensible name for this variable may be "waitingForServiceConfig" which defaults to true.

Reply:
Done.